### DISCUSSION TITLE
Getting material properties for Dirichlet boundary condition

### DISCUSSION THREAD

**Discussion by starkekr** [idaholab/moose#26340] [Q&A Modules: General]
I have a Dirichlet BC that depends on material properties and am struggling to find a way to enforce this BC.

Is there a way in MOOSE to interpolate material properties from elements to the nodes to apply Dirichlet BC's that depend on material properties?
Is something like this already implemented?

**Comment by starkekr** [comment]
seems like this is related to #23634

**Comment by GiudGiud** [comment]
Hello

Does the material property depend on the nonlinear variable?

A simple-ish scheme but that will be inexact would be:
- Use materialRealAux to project the matprop to a constant monomial auxvariable
- use a projectionAux to project the constant monomial on a lagrange / nodal  auxvariable
- use the nodal auxvariable in a MatchedValueBC

Guillaume
